openslides_backend/action/motion_category/number_motions.py

```diff
@@ -0,0 +1,225 @@
+        )
+        gmr_motions = GetManyRequest(
+            Collection("motion"),
+            meeting.get("motion_ids", []),
+            [
+                "lead_motion_id",
+                "meeting_id",
+                "category_weight",
+                "category_id",
+                "number",
+            ],
+        )
+        gmr_meetings = GetManyRequest(
+            Collection("meeting"),
+            [category.get("meeting_id", 1)],
+            [
+                "motions_number_with_blank",
+                "motions_number_min_digits",
+                "motions_amendments_prefix",
+            ],
+        )
+        result = self.datastore.get_many([gmr_categories, gmr_motions, gmr_meetings])
+        self.mem_categories = result.get(Collection("motion_category"), {})
+        self.mem_motions = result.get(Collection("motion"), {})
+        self.mem_meetings = result.get(Collection("meeting"), {})
+
+    def get_prefix(self, category_id: int) -> str:
+        """Get the prefix of a category. If none, get the prefix of the parent if exists."""
+        category = self.mem_categories.get(category_id, {})
+        if category.get("prefix"):
+            return category["prefix"]
+        elif not category.get("parent_id"):
+            return ""
+        else:
+            return self.get_prefix(category["parent_id"])
+
+    def get_lead_motion_id(self, motion_id: int) -> Optional[int]:
+        """Helper to get the lead_motions_id."""
+        motion = self.mem_motions.get(motion_id, {})
+        return motion.get("lead_motion_id")
+
+    def get_meeting_id(self, motion_id: int) -> Optional[int]:
+        """Helper to get the meeting_id from a motion."""
+        motion = self.mem_motions.get(motion_id, {})
+        return motion.get("meeting_id")
+
+    def get_affected_categories(self, category_id: int) -> List[int]:
+        """Get all affected categories, inits and calls the helper."""
+        affected_categories: List[int] = []
+        queue = [category_id]
+        self.helper_affected_categories(queue, affected_categories)
+        return affected_categories
+
+    def helper_affected_categories(
+        self, queue: List[int], aff_categories: List[int]
+    ) -> None:
+        """Fill the aff_categories, tree walk by level."""
+        if not queue:
+            return
+        aff_categories.append(queue[0])
+        category = self.mem_categories.get(queue[0], {})
+        queue.extend(self.sort_category_children(category.get("child_ids", [])))
+        self.helper_affected_categories(queue[1:], aff_categories)
+
+    def sort_category_children(self, child_ids: List[int]) -> List[int]:
+        """Sort the categories by weight. Important for the helper_affected_categories."""
+        weighted_child_ids = []
+        for id_ in child_ids:
+            category = self.mem_categories.get(id_, {})
+            weighted_child_ids.append((category.get("weight", 10000), id_))
```

Comment:
Don't use defaults in code like this. This field has to be set

Reply:
Okay.